### ISSUE TITLE
Dialogs should be phetioState:true

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#634]
From https://github.com/phetsims/phet-io/issues/1663, it made little sense to have Dialogs be phetioState false when they have potentially stateful children. This gives the possibility to make the state engine quite a bit simpler, and was easy to do for AboutDialog over (in https://github.com/phetsims/sun/commit/5dcd04e209f5849900025be28ccea096fa218661 and https://github.com/phetsims/joist/commit/3962e424d5663dc872b96a459887bbb51656a11d). Let's make it happen for keyboardHelpDialog (already working, but should make sure), and optionsDialog.

**Comment by zepumph** [comment]
I found that the OptionsDialog could not easily be made to support state, because it has a `createContent` function that cannot be created and destroyed generally. 

We have gotten this far keeping OptionsDialog out of state, so I think we can likely convince ourselves that it will be alright to have their children be unstateful until we need to change things.

**Comment by zepumph** [comment]
I have changed my mind since 16 minutes ago. OptionsDialog has supported state up until this point without actually looking like it has. `tryToCreatorAncestor` has back-doored around all of the structure we have set up to make state more explicit. 

Commit coming soon, but we should be better supporting options dialogs for state, since we already are supporting them.

**Comment by zepumph** [comment]
In the above commits I added support for OptionsDialog state even when tryToCreateAncestor wasn't recursive. 

Changes:
* I made sure that OptionsDialog content supported dispose correctly. This fixed assertions when the dialog is disposed and then reconstituted when setting state. 
* I made sure content was getting its tandem from the argument. This made sure that the phetioID was underneath `optionsDialog`, and as such we will be able to detect when options content is dynamic and set that flag appropriately. 

I will monitor CT for any other problems. As of the above commits, Dialogs now show up in state. Tagging @samreid, @chrisklus, and @KatieWoe so that they are aware of this fact.
